src/components/com_tjcertificate/administrator/libraries/certificate.php

```diff
@@ -941,4 +944,39 @@ protected function generateUniqueCertId($options)
 
 		return $certificateString;
 	}
+
+	/**
+	 * This function checks the certificate download permission 
+	 *
+	 * @param   STRING  $uniqueCertificateId  certificate Id
+	 *
+	 * @return  boolean
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function canDownload($uniqueCertificateId)
+	{
+		$user = Factory::getUser();
+
+		if (!$user->authorise('certificate.download.own', 'com_tjcertificate') && !$user->authorise('certificate.download.all', 'com_tjcertificate'))
+		{
+			return false;
+		}
+
+		if ($user->authorise('certificate.download.all', 'com_tjcertificate'))
+		{
+			return true;
+		}
+
+		if ($user->authorise('certificate.download.own', 'com_tjcertificate'))
+		{
+			$table = TJCERT::table("certificates");
+			$table->load(array('unique_certificate_id' => $uniqueCertificateId));
+
+			if ($user->get('id') == $table->user_id)
+			{
+				return true;
+			}
+		}
```

Comment:
Add return false; here at the end of the function

Reply:
done